src/v/storage/segment.h

```diff
@@ -41,10 +41,7 @@ class segment {
     struct offset_tracker {
         offset_tracker(model::term_id t, model::offset base)
           : term(t)
-          , base_offset(base)
-          , committed_offset(base)
-          , dirty_offset(base)
-          , stable_offset(base) {}
+          , base_offset(base) {}
```

Comment:
what values other offsets will have?

Reply:
they will be set to `std::numeric_limit<>::min()` i.e. very large negative number

Reply:
Why not set the committed offset.

Reply:
you mean set explicitly to uninitialized ?

Reply:
no i mean the `offset_tracker::commited_offset` field.

Reply:
feel free to disregard thsi comment. I think @dotnwat & @rystsov should review carefully as ... effectively all of storage depends on these offsets.

Reply:
I guess I don't understand this change either. Are the comments on the members below still correct?

Reply:
Segment dirty offset is the offset of last record of last batch in the segment. If we initialized the segment committed offset or dirty offset with `last offset + 1` we automatically make it to be first offset of a batch which is incorrect.

Reply:
Those comments from bellow are now in align with the code.